go-controller/pkg/ovn/egressip.go

```diff
@@ -2521,23 +2629,23 @@ func (e *EgressIPController) getNextHop(ni util.NetInfo, egressNodeName, egressI
 				mgmtPortAddressesStr := strings.Fields(mgmtPortAddress)
 				mgmtPortIP := net.ParseIP(mgmtPortAddressesStr[1])
 				if isEgressIPv6 && utilnet.IsIPv6(mgmtPortIP) {
-					nextHopIP = mgmtPortIP.To16().String()
+					nextHop = mgmtPortIP.To16().String()
 				} else {
-					nextHopIP = mgmtPortIP.To4().String()
+					nextHop = mgmtPortIP.To4().String()
 				}
 			}
 		}
-	} else if config.OVNKubernetesFeature.EnableInterconnect {
+	} else if config.OVNKubernetesFeature.EnableInterconnect && ni.TopologyType() == types.Layer3Topology {
```

Comment:
So this logic is saying if the egress node is remote, then use the transit switch IP as the next hop, but only for Layer 3. Why wouldn't this also be true for Layer 2?

Reply:
Not sure I get what you mean - there is no transit switch for a L2 network.
I added the extra constraint( ` && ni.TopologyType() == types.Layer3Topology {`) to increase code readability. Its a redundant constraint if we only process L3s or L2s networks.

Reply:
true, no transit switch ip for l2 network. I tried to clarify in my earlier comment in this file what I'm trying to say.